DOCUMENTATION.md

```diff
@@ -0,0 +1,115 @@
+resource/data source, along with the details of its schema.
+
+If you do not write a template for a resource or data source that you
+create, `make generate-docs` will use a basic template when it
+generates documentation for the provider. This template will include
+only the name of the resource/data source and the details of its
+schema. For resources in particular, this is insufficient to guide
+users as it does not include examples of HCL which make use of the
+resource, or instructions for how to import an existing resource into
+the Terraform state.
+
+Note: The documentation generation tool gathers the list of known data
+sources and resources from the provider itself, not by scanning the
+source directory. As a result, if you have created a new data source
+or resource but you have not added into the top-level schema in
+[provider.go](fastly/provider.go) it will not be included in the
+generated documentation. If you create a template for the new data
+source or resources and run `make validate-docs` the validator will
+produce an error as it will be unable to find the new data source or
+resource. The simplest way to avoid this problem is to write and
+execute the acceptance tests for the new data source or resource
+*before* working on its documentation; in order for the acceptance
+tests to pass the new data source or resource must already be included
+in the top-level schema.
+
+Before submitting a PR with your new data source or resource, you
+should add a proper template for it, and also ensure that the schema
+within the data source or resource has been properly documented as
+well. Review the following three sections for details.
+
+## Data Sources
+
+Each data source should have a template file and an example file
+showing basic usage of the data source in HCL. The template for [NGWAF
+Workspaces](templates/data-sources/ngwaf_workspaces.md.tmpl) can be
+used as a reference.
+
+## Resources
+
+Each resource should have a template file and at least two example
+files: one showing basic usage of the data source in HCL, and the
+other demonstrating how to import an existing resource into the
+Terraform state. The examples should be fully functional: they should
+include all resources necessary to create the resource being
+documented. The template for [Config
+Stores](templates/resources/configstore.md.tmpl) can be used as a
+reference.
+
+## Schema Attributes
+
+Every attribute in every schema must include a description. The
+description should provide all of the information necessary for a user
+to understand the purpose and usage of that attribute.
+
+To accomplish this, follow these steps:
+
+1. Use the description (from the public documentation) of the
+   corresponding attribute of the API resource that the schema
+   represents, whenever possible.
+
+1. If the attribute allows only a limited set of values (an
+   enumeration), include a sentence such as "Accepted values are
+   [`value 1`, `value 2`, `value 3`]".
```

Comment:
Use `code formatting` for attribute values and resource names throughout (which you mostly do).

For example, change:

"Accepted values are [value 1, value 2, value 3]."
  
  to:
  
"Accepted values are: value1, value2, value3."
    (Cleaner and more standard in Terraform docs.)

Reply:
Ahh, I think this is because of the square brackets... a Markdown renderer is likely to misintepret them. I'll fix the guide, but we will also need to find all the places in the provider where we are using that syntax now and correct them.

Reply:
be7cebb2